### DISCUSSION TITLE
Connect to a bluetooth device that requires a pin.

### DISCUSSION THREAD

**Discussion by aiacovella** [hbldh/bleak#1519] [Q&A]
I'm unsure of how exactly to handle a connection to a device that requires a pin to pair.  When using the library and connecting, will the underlying os handle it and prompt me for the pin or is that something I have to write and if so is there an example of it anywhere?

**Comment by dlech** [comment]
https://bleak.readthedocs.io/en/latest/api/client.html#pairing-bonding
